docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,454 @@
+    fn validator(&self, val_id: ValID) -> Option<Self::Validator>;
+}
+```
+
+Now for each vote in the commit, we can check if the validator existed in our
+trusted validator set, and thus if +1/3 of the trusted validators signed the new
+commit. We can use a function with the same signature:
+
+```rust
+fn verify_commit_trusting<V, C>(vals: V, commit: C) -> Result<(), Error>
+where
+    V: ValidatorSet,
+    C: Commit,
+{
+```
+
+### Validation
+
+Most of the above is about checking the signatures and voting power in commits, 
+but we also need to perform other validation checks, 
+like that the validator set hashes match what are in the header, and the lite
+client's trusted state actually hasn't expired. Pure functions for all of these
+checks should be provided.
+
+Some things are left explicitly unvalidated as they have minimal bearing on the correctness of the lite client.
+These include:
+
+- LastCommitHash
+	- In the skipping case, it's not possible to verify the header refers to the correct previous block without reverting to the sequential case. So in the sequential case, we don't validate this either. If it's incorrect, in indicates the validators are misbehaving, though we can only detect it as a lite client if there's a fork.
+- BlockID
+	- As mentioned, this includes a merkle root of the entire block and is not verifiable without downloading the whole block, which would defeat the purpose of the lite client!
+- Time
+	- Verifying the time would require us to download the commit for the previous block, and to take the median of the timestamps from that commit. This would add significant overhead to the lite client (an extra commit to validate for every block!). If the time is incorrect, in indicates that the validators are explicitly violating the protocol rules in a detectable way which full nodes should detect in the first place and shouldn't forward to lite clients, so there would probably be bigger issues at foot.
+
+There are likely a few other instances of things the lite client is not validating that it in theory could but likely indicate some larger problem afoot that the client can't do anything about anyways. Hence we really only focus on the correctness of commits and validator sets and detecting forks!
+
+### API
+
+While the library should include pure functions for all the forms of
+verification, it may not be right to expose all this functionality, since it may
+then be possible to use it incorrectly (ie. a header verifies correctly but the state is actually expired).
+
+Thus, a stateful API should be provided for managing the trusted state of a lite
+node. The trusted state is:
+
+```rust
+pub struct TrustedState<H, V>
+where
+    H: Header,
+    V: ValidatorSet,
+{
+    pub last_header: H, // height H-1
+    pub validators: V,  // height H
+}
+```
+
+And the main method for verifying and updating the state should look like:
+
+```rust
+pub fn verify_sequential<H, V, C>(&mut self, 
```

Comment:
The method's name and signature does not reflect that it also updates the (trusted) state. Which slightly confusing with the sentence in front of this snippet.